bin/varnishd/cache/cache_obj.c

```diff
@@ -220,17 +222,33 @@ void
 ObjExtend(struct worker *wrk, struct objcore *oc, ssize_t l, int final)
 {
 	const struct obj_methods *om = obj_getmethods(oc);
+	AN(om->objextend);
 
 	CHECK_OBJ_NOTNULL(wrk, WORKER_MAGIC);
 	CHECK_OBJ_NOTNULL(oc->boc, BOC_MAGIC);
 	(void)final;
 	assert(l >= 0);
 
 	Lck_Lock(&oc->boc->mtx);
-	AN(om->objextend);
 	if (l > 0) {
+		/* Pass prefetch applies when only sending to one client */
+		if (oc->flags & OC_F_PRIVATE) {
+			/* delivered_so_far will be re-read due to memory clobber
+			   from the CondWait opaque function call */
+			while (!(oc->flags & OC_F_ABANDON) &&
+				oc->boc->pass_prefetch > 0 &&
+				oc->boc->fetched_so_far > oc->boc->delivered_so_far + oc->boc->pass_prefetch) {
+				/* Wait 30ms for client to catch up */
+				(void)Lck_CondWait(&oc->boc->cond, &oc->boc->mtx, VTIM_real() + .030);
```

Comment:
I can live with it for the time being, but we might want to revisit later. I still think the timeout should not be necessary

Reply:
Maybe something on the scale of `idle_send_timeout` to still have a safety net but much less lock operations?

Reply:
I've changed this to 2.5 seconds, just to avoid kernel round-trips on slow clients.